provider/azureProvider.js

```diff
@@ -567,49 +564,48 @@ class AzureProvider {
+      fs.writeFileSync(path.join(this.serverless.config.servicePath, functionName+'-function.json'), JSON.stringify(functionJSON, null, 4));
+      resolve();
+    });
+  }
+
+  uploadFunction (functionName) {
     return new BbPromise((resolve, reject) => {
-      this.serverless.cli.log(`Packaging function: ${functionName}`);
-      const folderForJSFunction = path.join(functionsFolder, functionName);
-      const handlerPath = path.join(this.serverless.config.servicePath, filePath);
+      this.serverless.cli.log(`Uploading function: ${functionName}`);
 
-      if (!fs.existsSync(functionsFolder)) {
-        fs.mkdirSync(functionsFolder);
-      }
+      var functionZipFile = ''; 
 
-      if (!fs.existsSync(folderForJSFunction)) {
-        fs.mkdirSync(folderForJSFunction);
+      if (this.serverless.service.functions[functionName].package.artifact) {
+        functionZipFile = this.serverless.service.functions[functionName].package.artifact; 
+      } else if (this.serverless.service.artifact) {
+        functionZipFile = this.serverless.service.artifact; 
+      } else {
+        reject('Could not find zip package');
       }
-      fse.copySync(handlerPath, path.join(folderForJSFunction, 'index.js'));
-      const functionJSON = params.functionsJson;
 
-      functionJSON.entryPoint = entryPoint;
-      fs.writeFileSync(path.join(folderForJSFunction, 'function.json'), JSON.stringify(functionJSON, null, 4));
-      const functionZipFile = path.join(functionsFolder, functionName + '.zip');
-      zipFolder(folderForJSFunction, functionZipFile, function (createZipErr) {
-        if (createZipErr) {
-          reject(createZipErr);
-        } else {
-          const requestUrl = `https://${functionAppName}${config.scmZipApiPath}/${functionName}/`;
-          const options = {
-            url: requestUrl,
-            headers: {
-              Authorization: config.bearer + principalCredentials.tokenCache._entries[0].accessToken,
-              Accept: '*/*'
-            }
-          };
-
-          fs.createReadStream(functionZipFile)
-            .pipe(request.put(options, (uploadZipErr, uploadZipResponse) => {
-              if (uploadZipErr) {
-                reject(uploadZipErr);
-              } else {
-                fse.removeSync(functionZipFile);
-                resolve(uploadZipResponse);
-              }
-            }));
+      const requestUrl = `https://${functionAppName}${config.scmZipApiPath}/${functionName}/`;
+      const options = {
+        url: requestUrl,
+        headers: {
+          Authorization: config.bearer + principalCredentials.tokenCache._entries[0].accessToken,
+          Accept: '*/*'
         }
-      });
+      };
+
+      fs.createReadStream(functionZipFile)
+        .pipe(request.put(options, (uploadZipErr, uploadZipResponse) => {
+          if (uploadZipErr) {
+            reject(uploadZipErr);
+          } else {
+            // fse.removeSync(functionZipFile);
```

Comment:
Can this line be removed? Now that the Serverless Framework handles creating the zip file, do they also clean it up?